src/Dfc.CourseDirectory.Core/Search/AzureCognitiveSearch/AzureCognitiveSearchClient.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Search.Documents;
+
+namespace Dfc.CourseDirectory.Core.Search.AzureCognitiveSearch
+{
+    public class AzureCognitiveSearchClient<TQuery, TResult> : ISearchClient<TQuery, TResult>
+        where TQuery : class, IAzureCognitiveSearchQuery
```

Comment:
Are the `class` constraints required here?

Reply:
Yes and no. It'll work without them, but the behaviour could be unpredictable. I'll take them out as there's no need for them at this time.